### DISCUSSION TITLE
Problem with extruder heating

### DISCUSSION THREAD

**Discussion by xXTronXx3971** [Jyers/Marlin#270] [Q&A] [answered]
I was enjoying this firmware but I had to get a replacement 4.2.7 board because of the layer shift issue. Once I installed that board and flashed the 4.2.7 version of Jyers firmware it gives me an error (Printer Kill Reason: Heating failed Restart Required) and beeps like crazy. I thought this might have been my thermistor going bad but I flashed crealitys bl firmware and it works fine without that error.

Edit meant hotend not extruder

**Comment by Jyers** [comment]
Does it throw this error right away, or did it happen as a result of some operation? If you could give a rundown of how exactly this occurred that works be great. It sounds to me like a configuration issue, are you compiling you're own or using one of the precompiled configs?

**Reply by xXTronXx3971** [reply]
Basically, whenever I would start a print with abl it would get to the first point, measure it once, then throw out the error code. I was using the precompiled firmware that you provide but oddly enough when I reflashed your firmware again after reformatting the sd card and using the exact same file it seems to be working completely fine. My best guess at the moment is that it was possibly a corrupted sd card.

**Comment by Schnabulation** [comment]
Funnily enough after flashing the latest version yesterday I also had a M112 "Heating failed" - the very first time I have seen a M112 error. Never had something remotely close to a M112 so far. And @Jyers you know I'm using your firmware for a long time with the same exact configs - never had a problem.

I have no idea if it's related to recent changes (I checked your commits since my last merge and could not find anything related) but I seriously never had a problem.

For me it happened when I preheated to PETG - it preheated nearly 80% up and then threw the error. Didn't happen since...
If it does I'll report back.

**Comment by Jyers** [comment, marked as answer]
I looked into it a bit since I was interested if anything could be tune to help prevent this. I found that this is dependent on WATCH_TEMP_PERIOD and WATCH_TEMP_INCREASE. I think I will increase the watch temp period slightly to reduce the frequency of this error.

**Reply by Schnabulation** [reply]
> WATCH_TEMP_PERIOD and WATCH_TEMP_INCREASE

If I get this right, these are settings in configuration_adv.h and nothing hardcoded into a *.cpp file, correct? I might play with this as well...

**Reply by Schnabulation** [reply]
Futhermore: My settings are
`#define WATCH_TEMP_PERIOD 40`
`#define WATCH_TEMP_INCREASE 2`
This would mean that the nozzle hasn't increased the temperature by at least 2° in a timespan of 40 seconds. If the printer can not do this there is another problem laying beyond...

**Reply by Jyers** [reply]
While 40 seconds for 2 degrees seems like it should always be fine, it will keep checking every 40 seconds, so there is a chance that if it is in the very beginning or end of the process, this could trip. It's unlikely, but that is why you would only see this error once in a blue moon. Either way increasing this slightly should change the occurrence rate from extremely rarely to pretty much never.
